src/k8s/pkg/utils/k8s/status.go

```diff
@@ -9,24 +9,30 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// WaitApiServerReady waits until the kube-apiserver becomes available.
+func (c *Client) listNodes(ctx context.Context) (*v1.NodeList, error) {
+	nodes, err := c.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to list nodes: %v", err)
+	}
+	return nodes, nil
+}
```

Comment:
Let's remove this, as it is only wrapping one line of useful code: `c.CoreV1().Nodes().List(ctx, metav1.ListOptions{})`, without any other transformation or error handling.